src/transformStateWithClones.js

```diff
@@ -7,7 +7,46 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let previousState = state;
```

Comment:
The variable `previousState` is assigned directly to the input `state` object, which risks mutating the original `state` if not handled properly. Instead, create a copy of the `state` before manipulating it to avoid any side effects.